src/data.rs

```diff
@@ -0,0 +1,272 @@
+        }
+    }
+
+    /// Allow actions through a main button
+    ///
+    /// # Example:
+    ///
+    /// ```no_run
+    /// # use mac_notification_sys::*;
+    /// let _ = NotificationOptions::new().main_button(MainButton::SingleAction("Main button"));
+    /// ```
+    pub fn main_button(&mut self, main_button: MainButton<'a>) -> &mut Self {
+        self.main_button = Some(main_button);
+        self
+    }
+
+    /// Display a close button with the given name
+    ///
+    /// # Example:
+    ///
+    /// ```no_run
+    /// # use mac_notification_sys::*;
+    /// let _ = NotificationOptions::new().close_button("Close");
+    /// ```
+    pub fn close_button(&mut self, close_button: &'a str) -> &mut Self {
+        self.close_button = Some(close_button);
+        self
+    }
+
+    /// Display an icon on the left side of the notification
+    ///
+    /// NOTE: The icon of the app associated to the bundle will be displayed next to the notification title
+    ///
+    /// # Example:
+    ///
+    /// ```no_run
+    /// # use mac_notification_sys::*;
+    /// let _ = NotificationOptions::new().app_icon("/path/to/icon.icns");
+    /// ```
+    pub fn app_icon(&mut self, app_icon: &'a str) -> &mut Self {
+        self.app_icon = Some(app_icon);
+        self
+    }
+
+    /// Display an image on the right side of the notification
+    ///
+    /// # Example:
+    ///
+    /// ```no_run
+    /// # use mac_notification_sys::*;
+    /// let _ = NotificationOptions::new().content_image("/path/to/image.png");
+    /// ```
+    pub fn content_image(&mut self, content_image: &'a str) -> &mut Self {
+        self.content_image = Some(content_image);
+        self
+    }
+
+    /// Set an identifier to group multiple notifications together
+    /// Notifications in group will dismiss each other so only the latest one is displayed
+    ///
+    /// # Example:
+    ///
+    /// ```no_run
+    /// # use mac_notification_sys::*;
+    /// let _ = NotificationOptions::new().group_id("my_group_id");
+    /// ```
+    pub fn group_id(&mut self, group_id: &'a str) -> &mut Self {
+        self.group_id = Some(group_id);
+        self
+    }
+
+    /// Schedule the notification to be delivered at a later time
+    ///
+    /// The delivery_date parameter is the time at which to schedule the notification
+    ///
+    /// The synchronous parameter defines whether or not the notification
+    /// should wait synchronously until it has been delivered (if no action is set)
+    ///
+    /// # Example:
+    ///
+    /// ```no_run
+    /// # use mac_notification_sys::*;
+    /// # use chrono::offset::*;
+    /// let stamp = Utc::now().timestamp() as f64 + 5.;
+    ///
+    /// // Synchronous is true, this will wait until the user
+    /// // interacts with the notification before returning
+    /// let _ = NotificationOptions::new().delivery_date(stamp, true);
+    /// ```
+    pub fn delivery_date(&mut self, delivery_date: f64, synchronous: bool) -> &mut Self {
```

Comment:
What if the user sets both `delivery_date` and `delivery_date_sync`? Should it just use whichever is latest? 

Or do you mean that the `delivery_date_sync` should just be for the bool part? If so, then what happens if `delivery_date_sync` is set, but not `delivery_date`?

Reply:
the user can already call `delivery_date` twice with conflicting inputs, but you're right, it's not super clear
but I just see that you even store it in the same property as an anonymous tuple
things like this: https://github.com/PandawanFr/mac-notification-sys/blob/master/src/data.rs#L211
I don't think there is a need to mix up both these concepts, and the conciseness of this tuple doesn't even save any space/performance. It's just harder to understand. How about just making the syncness it's own thing?

pick a reasonable default and add an extra property (e.g. `[a]sync_delivery`/ `.[a]sync_delivery()`)

Reply:
Yeah so this links back to my original point at the very start
> I'm thinking of adding a synchronous option instead of forcing it with buttons and scheduling. I saw that #4 mentioned issues with waiting synchronously with a while loop so maybe this would be a good fix? But it also seems kind of wrong to allow for actions and yet they do nothing unless you set the "synchronous" option.

The way I have it, if you ask for actions (aka buttons) it will automatically use synchronous mode. But I chose to give the option for synchronous delivery when scheduling for later. Maybe what I could do is, by default it's always synchronous, and I could add an `asynchronous()` option to be more explicit? (Since reasoning about it synchronously makes more sense and you can "opt in" for the potentially unwanted async behavior).

Reply:
sounds good to me